src/groundlight/client.py

```diff
@@ -538,10 +578,12 @@ def wait_for_confident_result(
         :return: ImageQuery
         :rtype: ImageQuery
         """
+        if confidence_threshold is None:
+            if isinstance(image_query, str):
+                image_query = self.get_image_query(image_query)
+            confidence_threshold = self.get_detector(image_query.detector_id).confidence_threshold
 
-        def confidence_above_thresh(iq):
-            return iq_is_confident(iq, confidence_threshold=confidence_threshold)
-
+        confidence_above_thresh = partial(iq_is_confident, confidence_threshold=confidence_threshold)
```

Comment:
thanks for teaching me about `partial` - this is very haskell-y.

Reply:
Haha its honestly really nice sometimes. I love it because its a bit of a functional holdover